File: web_app/lib/features/score_search/models/score.dart
import 'package:score/features/score_search/models/part.dart';

class Score {
  Score({
    required this.title,
    required this.arrangementName,
    required this.alsoKnownAs,
    required this.composers,
    required this.lyricists,
    required this.parts,
  });

  final String title;
  final String? arrangementName;
  final List<String> alsoKnownAs;
  final List<String> composers;
  final List<String> lyricists;
  final List<Part> parts;

  Set<String> get creators => {...composers, ...lyricists};
  Set<String> get instruments =>
      parts.expand((part) => part.instruments).toSet();
}

File: backend/internal/search/score.go
package search

import "score/backend/pkgs/models"

func ScoreToDocument(score *models.Score, id string) map[string]interface{} {
	if score == nil {
		return nil
	}
	return map[string]interface{}{
		"id":          id,
		"title":       score.Title,
		"composers":   score.Composers,
		"lyricists":   score.Lyricists,
		"instruments": score.Instruments(),
	}
}
